common/functions/src/scalars/maths/exp.rs

```diff
@@ -14,62 +14,64 @@
 
 use std::fmt;
 
-use common_datavalues::prelude::*;
-use common_datavalues::DataTypeAndNullable;
+use common_datavalues2::prelude::*;
 use common_exception::ErrorCode;
 use common_exception::Result;
 
-use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::cast_column_field;
+use crate::scalars::function2_factory::Function2Description;
 use crate::scalars::function_factory::FunctionFeatures;
-use crate::scalars::Function;
+use crate::scalars::Function2;
 
 #[derive(Clone)]
 pub struct ExpFunction {
     _display_name: String,
 }
 
 impl ExpFunction {
-    pub fn try_create(_display_name: &str) -> Result<Box<dyn Function>> {
+    pub fn try_create(_display_name: &str) -> Result<Box<dyn Function2>> {
         Ok(Box::new(ExpFunction {
             _display_name: _display_name.to_string(),
         }))
     }
 
-    pub fn desc() -> FunctionDescription {
-        FunctionDescription::creator(Box::new(Self::try_create))
+    pub fn desc() -> Function2Description {
+        Function2Description::creator(Box::new(Self::try_create))
             .features(FunctionFeatures::default().deterministic().num_arguments(1))
     }
 }
 
-impl Function for ExpFunction {
+impl Function2 for ExpFunction {
     fn name(&self) -> &str {
         &*self._display_name
     }
 
-    fn return_type(&self, args: &[DataTypeAndNullable]) -> Result<DataTypeAndNullable> {
-        let nullable = args.iter().any(|arg| arg.is_nullable());
-
-        let data_type = if args[0].is_numeric() || args[0].is_string() || args[0].is_null() {
-            Ok(DataType::Float64)
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        let data_type = if args[0].data_type_id().is_numeric()
+            || args[0].data_type_id().is_string()
```

Comment:
We should strict args[0] to be `args[0].data_type_id().is_numeric()`.

1. Reject other types like: String 
2. `passthrough_null` is enabled by default, so null_type is covered in `Function2Adaptor`:   https://github.com/datafuselabs/databend/blob/f076522bd3e12c4e3a8ef59c3a6e949446e4adb9/common/functions/src/scalars/function2_adapter.rs

Reply:
Okay, so we will no longer support `exp('2')` and I will remove it from stateless.

Reply:
or change  to `SELECT exp('2'); -- {ErrorCode 1007}`

Reply:
I prefer the latter.